File: Course Materials/Examples and Templates/OpenMP_cours/cours_ex6.f90
program parallel

! Dans une procédure, toutes les variables transmises par argument 
!(dummy parameters) héritent du statut défini dans 
!l'étendue lexicale (statique) de la région.

 implicit none

 integer :: a, b

 a = 92000

 !$OMP PARALLEL SHARED(a) PRIVATE(b)
    call sub(a, b)
    print *,"B vaut : ",b
 !$OMP END PARALLEL

end program parallel


subroutine sub(x, y)
 use OMP_LIB
 implicit none
 integer, intent(in)  :: x
 integer, intent(out) :: y

 y = x + OMP_GET_THREAD_NUM()
end subroutine sub

File: Course Materials/Examples and Templates/OpenMP_cours/cours_ex6.c
#include <stdio.h>
#include <omp.h>

/* 
Dans une procédure, toutes les variables transmises par argument 
(dummy parameters) héritent du statut défini dans 
l'étendue lexicale (statique) de la région.
*/

int main()
{
  void sub(int x, int *y);
  int a, b;


  a = 92000;
#pragma omp parallel shared(a) private(b)
  {
    sub(a, &b);
    printf("b vaut : %d\n",b);
  }
  return 0;
}
          



void sub(int x, int *y)
{
  *y = x + omp_get_thread_num();
}
